Source/BugsnagDeviceWithState.m

```diff
@@ -75,6 +75,33 @@ - (instancetype)init {
     return self;
 }
 
++ (BugsnagDeviceWithState *) deviceFromJson:(NSDictionary *)json {
+    BugsnagDeviceWithState *device = [BugsnagDeviceWithState new];
+    device.id = json[@"id"];
+    device.freeMemory = json[@"freeMemory"];
+    device.freeDisk = json[@"freeDisk"];
+    device.locale = json[@"locale"];
+    device.manufacturer = json[@"manufacturer"];
+    device.model = json[@"model"];
+    device.modelNumber = json[@"modelNumber"];
+    device.orientation = json[@"orientation"];
+    device.osName = json[@"osName"];
+    device.osVersion = json[@"osVersion"];
+    device.runtimeVersions = json [@"runtimeVersions"];
+    device.totalMemory = json[@"totalMemory"];
+
+    id jailbroken = json[@"jailbroken"];
+    if (jailbroken) {
+        device.jailbroken = [(NSNumber *) jailbroken boolValue];
```

Comment:
Jailbroken is not listed as nullable - do we need a default?

Reply:
The default will be false for both the `jailbroken` and `inForeground` properties as that's the default value for the `BOOL` type in Objective-C.